docs/user/tutorials/vote-extensions/mitigate-frontrunning.md

```diff
@@ -0,0 +1,407 @@
+# Mitigate Auction Front-Running
+
+## What are Vote extensions?
+
+Vote extensions is arbitrary information which can be inserted into a block. This feature is part of ABCI 2.0, which is available for use in the SDK 0.50 release and part of the 0.38 CometBFT release. 
+
+More information about vote extensions can be seen [here](../../../build/building-apps/04-vote-extensions).
+
+## Context
+
+For this tutorial we are using an example of an application that is mitigating auction front-running. 
+
+Please checkout https://github.com/fatal-fruit/abci-workshop to see the example.
+
+
+## Simulation of Auction Front-Running
+
+Please checkout the branch `part-1` of the example and then run:
+
+```shell
+./scripts/single_node/setup.sh
+```
+
+In another window run:
+
+```shell
+./scripts/single_node/frontrun.sh
+```
+
+Here we can see alice attempts to reserve the namespace `bob.cosmos`
+
+To verify the occurrence of front-running, navigate to the logs and search for instances of `Found a Bid to Snipe`. This specific message indicates the desired behavior, demonstrating that front-running has occurred.
+
+Below we query the namespace `bob.cosmos` to see if alice managed to successfully reserve `bob.cosmos`
+
+```shell 
+./build/cosmappd query ns whois bob.cosmos
+```
+
+You should receive something like this:
+
+```shell
+name:
+  amount:
+  - amount: "2000"
+    denom: uatom
+  name: bob.cosmos
+  owner: cosmos185gc7c296w0xjlq9kjdt6gghnqvdmyckv64e7a
+  resolve_address: cosmos185gc7c296w0xjlq9kjdt6gghnqvdmyckv64e7a
+```
+
+The owner address corresponds to Alice (`val1` in the keyring)
+
+## Mitigating Front-running with Vote Extensions Steps
+
+
+To initiate the process, we introduce new types within the `abci/types.go` file. Carefully consider the specific requirements for each type, bearing in mind that we utilise our own mempool for transaction ordering.
+
+```go
+package abci
+
+import (
+	//import the necessary files
+)
+
+type PrepareProposalHandler struct {
```

Comment:
am i suppose to copy this over? 

if this is an advanced tutorial we should explain the structs so people understand why its needed